src/Repository/CreditRepository.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace App\Repository;
+
+use Wikimedia\ToolforgeBundle\Service\ReplicasClient;
+
+class CreditRepository {
+
+	public function getDatabaseName( $domain, $family ) {
+		if ( $family === 'wikisource' ) {
+			if ( in_array( $domain, [ 'old', 'www', 'mul' ] ) ) {
+				$dbname = 'sourceswiki_p';
+			} elseif ( $domain === 'zh-min-nan' ) {
+				$dbname = 'zh_min_nanwikisource_p';
+			} else {
+				$dbname = $domain . $family . '_p';
+			}
+		} else {
+			$dbname = $domain . $family . '_p';
+		}
+		return $dbname;
+	}
+	
+	public function getPageCredits( ReplicasClient $replicasClient, $lang, $namespaces, $pages ) : array {
+		$dbname = $this->getDatabaseName( $lang, 'wikisource' );
+
+		$conn = $replicasClient->getConnection( $dbname );
+		$pageNames = [];
+		$namespaceName = "";
+		$splittedPages = [];
+	
+		foreach ( $pages as $page ) {
+			$splitPage = explode( ":", $page, 2 );
+			if ( isset( $splitPage[1] ) ) {
+				$namespaceName = $splitPage[0];
+				$pageNames[] = $splitPage[1];
+				$namespaceId = $namespaces[$namespaceName];
+				$splittedPages[$namespaceId][] = "'" . $splitPage[1] . "'";
```

Comment:
This will presumably cause an SQL error for pages with an apostrophe in the title. I usually use Doctrine's `Connection::quote()` method for this, so something like:
```
$splittedPages[$namespaceId][] = $conn->quote($splitPage[1])
```

Reply:
Done. Thank you!